jetcd-core/src/test/java/io/etcd/jetcd/AuthClientTest.java

```diff
@@ -15,262 +15,158 @@
  */
 package io.etcd.jetcd;
 
-import static com.google.common.base.Charsets.UTF_8;
+import static io.etcd.jetcd.TestUtil.bytesOf;
 import static org.assertj.core.api.Assertions.assertThatThrownBy;
-import static org.assertj.core.api.AssertionsForClassTypes.assertThat;
+import static org.assertj.core.api.Assertions.assertThat;
 
 import io.etcd.jetcd.auth.AuthRoleGetResponse;
 import io.etcd.jetcd.auth.AuthRoleListResponse;
 import io.etcd.jetcd.auth.Permission;
 import io.etcd.jetcd.auth.Permission.Type;
-import io.etcd.jetcd.launcher.EtcdCluster;
-import io.etcd.jetcd.launcher.EtcdClusterFactory;
-import java.io.IOException;
+import io.etcd.jetcd.launcher.junit.EtcdClusterResource;
+
 import java.net.URI;
 import java.util.List;
-import java.util.concurrent.ExecutionException;
-import org.testng.annotations.AfterTest;
-import org.testng.annotations.BeforeTest;
-import org.testng.annotations.Test;
+
+import org.junit.BeforeClass;
+import org.junit.ClassRule;
```

Comment:
beforeClass and Classrule is a junit 4 thing right? I think we currently want to move to junit 5. So maybe this refactoring can leverage junit 5 syntax?

This test https://github.com/etcd-io/jetcd/blob/master/jetcd-core/src/test/java/io/etcd/jetcd/KVNamespaceTest.java uses junit 5. Maybe you can base it off that or come with something better>

Reply:
It is JUnit4 thing and I tried to refactor everything to JUnit5 but wasn't able to do so. I haven't found an easy way to convert `@Rule` to JUnit5 extension. As you mentioned before this PR already does 2 things so I didn't want to add one more.

Reply:
fair enough, would you like to convert this Junit 5 in another pr then?

Reply:
I would take another look on how to replace `@Rule`. If I would find a way - yes, I'll do another PR, if not - then I won't :)